frontend/components/songs/song_card_dropdown_item.jsx

```diff
@@ -32,9 +33,13 @@ const SongCardDropdownItem = ({
                 .then((playlistId) => history.push(`/playlist/${playlistId}`));
         } else if (depthLevel === 1) {
             updateSongCardDropdownState({ isOpen: false });
-            let playlist = playlists[index];
-            createNewPlaylisted(selectedSong.id, playlist.id)
-                .then((playlistId) => history.push(`/playlist/${playlistId}`)
+            let selectedPlaylist = playlists[index];
+            createNewPlaylisted(selectedSong.id, selectedPlaylist.id).then(
+                (selectedPlaylistId) => {
+                    if (currentItem.id === selectedPlaylistId) {
+                        displayPlaylist(selectedPlaylistId);
```

Comment:
We're already on the `selectedPlaylist` right? Does this just refresh the props?

Reply:
`selectedPlaylist` is the playlist the user is choosing from the dropdown. This may or may not be the same as the playlist in the current view. If it is the same, then we need to display (re-fetch and view) the playlist being currently viewed. If it is not the same, we can just re-fetch all the playlists on the side, which will update the playlists in the sidebar, but we don't need to immediately view the playlist (because the user is still currently viewing this one).

Reply:
I changed the generic `index` variable to `selectedIndex`